osu.Game/Input/GameIdleTracker.cs

```diff
@@ -1,17 +1,26 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using osu.Framework.Bindables;
 using osu.Framework.Input;
+using osu.Game.Configuration;
 
 namespace osu.Game.Input
 {
     public class GameIdleTracker : IdleTracker
     {
         private InputManager inputManager;
 
-        public GameIdleTracker(int time)
+        public GameIdleTracker(int time, SessionStatics statics)
```

Comment:
Rather than passing this in as a parameter, you should just bind to the already-exposed `IsIdle` bindable.

So in `OsuGame` you'd have something like this:

```csharp
var sessionIdleTracker = new GameIdleTracker();
Add(sessionIdleTracker);
sessionIdleTracker.IsIdle.BindValueChanged(() => {
    if (e.NewValue)
        statics.ResetValues();
});
```

Note that bindable changed events only fire when a change actually occurs, so you don't need to check `NewValue!=OldValue` as you had.

Reply:
Aha! I thought I was missing something.

Makes sense, thank you.